exports/dataset.cpp

```diff
@@ -1040,6 +1040,10 @@ PYBIND11_MODULE(dataset, m) {
       // DatasetSlice matches the overload below for py::array_t. I have not
       // understood all details of this yet though. See also
       // https://pybind11.readthedocs.io/en/stable/advanced/functions.html#overload-resolution-order.
+      .def("__setitem__", [](Dataset &self, const std::string &name,
+                             DatasetSlice &slice) { self.insert(name, slice); })
+      .def("__setitem__", [](Dataset &self, const std::string &name,
+                             Dataset &slice) { self.insert(name, slice); })
```

Comment:
This relates to your comment on Slack: If implemented this way, you indeed to not need to change `SubsetHelper`. If we choose the alternate syntax:

```python
d.subset[name] = subset
```
you need to make changes similar to the ones here, but in `SubsetHelper`.

So the question is, which syntax is to be preferred? The corresponding `__getitem__` (`d['name']`) has been renamed to `d.subset['name']` for clarity recently. Having a `__setitem__` with no corresponding `__getitem__` is maybe a bit confusing? Not sure what the ultimate answer is.

Reply:
Good question. I have a slight preference for the way it's implemented here already as seems more intuitive. However from a API support point of view, the more specific syntax via the `Dataset.subset` would be easier to keep supporting or revoke if we later decided to go for the syntax as provided already.

Reply:
Maybe this highlights another unclear point: As it is, this would insert coordinates that are missing in the left-hand-side. Maybe this should be prevented? Or can it be clarified with better naming?